### ISSUE TITLE
Handle dunning accounts in a more proactive way

### ISSUE THREAD

**Issue by dabeeeenster** [SolidStateGroup/bullet-train-api#1822]
When a saas customer fails an invoice payment, we will receive a callback from Chargebee alerting us that the subscription is in dunning. When this happens show a banner similar to the linkedin one:

![image](https://user-images.githubusercontent.com/173290/210543659-45b0fd94-0358-4d40-ac48-e51adf4a1243.png)

If we then receive a successful payment webhook for the subscription, remove the banner. 

After 60 days the account will automatically be marked as cancelled by chargebee. When that happens, prevent the users from logging into the platform until we have received payment for the outstanding invoices.

## Acceptance Criteria

 - [ ] If a paid subscription fails to pay an invoice, the flagsmith dashboard displays a banner to that effect
 - [ ] When the subscription is cancelled by chargebee, users should not be able to log into the flagsmith dashboard

**Comment by kyle-ssg** [comment]
The frontend has the mechanism to show a butter bar / put the site into read only mode. It's currently powered by segments, reassign to me when the appropriate flag against the org is setup.

**Comment by kyle-ssg** [comment]
@dabeeeenster If possible could you put a test org into this segment and let me know if this is suitable?

**Comment by dabeeeenster** [comment]
OK it turns out we have the code to turn the UI into effectively read only mode. So we can use that, but we need to do the following on the API side: 

- Add a new column to the `organisation` entity called `subscription_status` that can take the following values:
    - `no_subscription` - for free/new accounts
    - `active` - for subscriptions without any overdue invoices
    - `in_dunning` - for subscriptions with unpaid invoices but within the 60 day dunning window
    - `cancelled` - for subscriptions that have been cancelled

We then need to listen to the appropriate chargebee webhooks and set the `subscription_status` based on the appropriate webhooks from chargebee.

**Comment by matthewelwell** [comment]
All organisations are now created with a default (free) subscription so we should store this data against the subscription rather than directly against the organisation.

**Comment by dabeeeenster** [comment]
Ah yes that's a better solution.

**Comment by zachaysan** [comment]
The backend of this is up for review, but we will still need @novakzaballa's help for the frontend tie-in

**Comment by zachaysan** [comment]
The API has been added to the existing `/api/v1/organisations/` endpoint where the following results:

```json
{"results": [{"subscription": {"id": 22, "billing_status": "DUNNING"}}]
```

(I've removed most of the post data noise)

If a specific org is dunning it should be highlighted. It could also be dunning in a different organisation if a user has multiple organisations. Possibly can design dunning across a non-active app, but I'm not sure about the workflow for that. Maybe just check the current organisation?
